### DISCUSSION TITLE
accessing same camera source for mediamtx and another camera source running

### DISCUSSION THREAD

**Discussion by sravich1** [bluenviron/mediamtx#3478] [Questions]
### Question

Hi,
I am running mediamtx which is configured for raspberrypi ID: 0 .
I have another python code which needs the same camera streamed output so that I can access for other tasks.
How can I access camera from mediamtx and provide the streamed output as an input to another task(even visa ve possible??)
Kindly through some light .
thanks
Ravi

**Comment by moenodedev** [comment]
The readme has instructions for Raspberry Pi Cameras that can help you 😄 
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#raspberry-pi-cameras

**Comment by aler9** [comment]
Hello, first you need to ingest the camera into MediaMTX:

```yml
paths:
  rpi:
    source: rpiCamera
```

Then you need to edit the Python script and pull the camera stream from MediaMTX instead of directly from the camera, for instance by using RTSP:

```python
vcap = cv.VideoCapture("rtsp://localhost:8554/rpi")
while(1):
    ret, frame = vcap.read()
    cv.imshow('VIDEO', frame)
    cv.waitKey(1)
```

This is because the library in charge of interacting with the Raspberry Pi camera (libcamera) doesn't support multiple accesses at once.

[a comment by an automated account is not shown]
